Fix variable scope issue in undo functionality.

Upstream commit 20c30f6d189eb78bf6c0eae4b5f4b9d5e1baf0ba.

Index: src/main.c
--- src/main.c.orig
+++ src/main.c
@@ -708,18 +708,20 @@ int main(int argc, char *argv[])
 				}
 				break;
 			case 'u': // Undo
-				move_t old_move;
-				if (undo_stack_pop(&old_move))
-				{	// Stack empty
+				{
+					move_t old_move;
+					if (undo_stack_pop(&old_move))
+					{	// Stack empty
+						break;
+					}
+					x = old_move.x;
+					y = old_move.y;
+					posy = (y-GRID_NUMBER_START_Y)/GRID_COL_DELTA;
+					posx = (x-GRID_NUMBER_START_X)/GRID_LINE_DELTA;
+					user_board[posy*9+posx] = old_move.prev_val;
+					fill_grid(user_board, plain_board, x, y);
 					break;
 				}
-				x = old_move.x;
-				y = old_move.y;
-				posy = (y-GRID_NUMBER_START_Y)/GRID_COL_DELTA;
-				posx = (x-GRID_NUMBER_START_X)/GRID_LINE_DELTA;
-				user_board[posy*9+posx] = old_move.prev_val;
-				fill_grid(user_board, plain_board, x, y);
-				break;
 
 			default:
 				break;
@@ -755,4 +757,3 @@ int main(int argc, char *argv[])
 	endwin();
 	return EXIT_SUCCESS;
 }
-
